mlton/front-end/ml.lex

```diff
@@ -442,17 +446,17 @@ real=(~?)(({decnum}{frac}?{exp})|({decnum}{frac}{exp}?));
 <TEXT>"\\\""     => (addTextString "\""; continue ());
 <TEXT>\\\\       => (addTextString "\\"; continue ());
 <TEXT>\\{ws}+    => (YYBEGIN TEXT_FMT; continue ());
-<TEXT>\\{eol}    => (Source.newline (source, yypos + 1); YYBEGIN TEXT_FMT; continue ());
+<TEXT>\\{eol}    => (Source.newline (source, newLinePos (yypos + 1) yytext); YYBEGIN TEXT_FMT; continue ());
```

Comment:
This should also be `newLinePos yypos yytext`; the `yypos + 1` skips over the `\` character.